File: solutions/javascript/word-count/1/word-count.js

export const countWords = (phrase='') => {
  let words = {};
  let words1;
  if (phrase.startsWith("'") && phrase.endsWith("'")){
    phrase = phrase.slice(1,-1);
  }
  words1 = phrase.replace(/[^a-zA-Z0-9']+/g,' ').toLocaleLowerCase()
    .trim().split(' ');
   for (const word1 of words1) {
       let word = word1;
       if (word.startsWith("'") && word.endsWith("'")) {
         word = word.slice(1,-1);
       }
       if (words[word]) {
         words[word]++;
       } else {
         words[word] = 1;
       }
   }
  return words;
};

File: solutions/lua/word-count/1/word-count.lua
local function count_words(s)
  local words = {}
  for word in (s or ""):lower():gmatch("[%a%d']+") do
    word = word:match("^'?(.-)'?$")
    if word and #word > 0 then
      words[word] = (words[word] or 0) + 1
    end
  end
  return words
end

return { count_words = count_words }